collector/otel-setup/traceparent.mdx

```diff
@@ -0,0 +1,126 @@
+which automatically appends comments (query tags) to SQL queries.
+You can enable this via Rails configuration in `config/application.rb` or an
+initializer:
+
+```ruby
+config.active_record.query_log_tags_enabled = true
+```
+
+By default, this doesn't have a traceparent query tag, so you'll need to add it
+manually. New comment tags can be defined by adding them in a `Hash` to the tags
+`Array`.
+You can append the traceparent tag like the following:
+
+```ruby
+config.active_record.query_log_tags = [
+  :application,
+  :controller,
+  :action,
+  :job,
+  {
+    traceparent: -> () {
+      current_span = ::OpenTelemetry::Trace.current_span
+      return if current_span == ::OpenTelemetry::Trace::Span::INVALID
+
+      span_context = current_span.context
+      trace_flag   = span_context.trace_flags.sampled? ? '01' : '00'
+      format(
+        '00-%<trace_id>s-%<span_id>s-%<trace_flags>.2d',
+        trace_id: span_context.hex_trace_id,
+        span_id: span_context.hex_span_id,
+        trace_flags: trace_flag
+      )
+    }
+  }
+]
+```
+
+With above configuration, queries will have the comments like the following:
+
+```sql
+SELECT "articles".* FROM "articles" /*application='myblog',action='index',controller='articles',traceparent='00-953c157ad37a81d94fe83bb95cfc8add-3cf784b99bd43910-01'*/
+```
+
+### Marginalia
+
+If you are using Rails 6 and below, you can attach SQL comments using the
+[marginalia](https://github.com/basecamp/marginalia) gem.
+
+You can define a transparent tag in the initializer
+(e.g. `config/initializers/marginalia.rb`):
+
+```ruby
+module Marginalia
+  module Comment
+    def self.traceparent
+      current_span = ::OpenTelemetry::Trace.current_span
+      return if current_span == ::OpenTelemetry::Trace::Span::INVALID
+
+      span_context = current_span.context
+      trace_flag   = span_context.trace_flags.sampled? ? '01' : '00'
+      format(
+        '00-%<trace_id>s-%<span_id>s-%<trace_flags>.2d',
+        trace_id: span_context.hex_trace_id,
+        span_id: span_context.hex_span_id,
+        trace_flags: trace_flag
+      )
+    end
+  end
+end
+
+Marginalia::Comment.components = [:application, :controller, :action, :job, :traceparent]
+```
+
+### Limitations
+
+If you're using the OpenTelemetry SDK with Ruby, the pganalyze collector
+integration is best used with `OpenTelemetry::Instrumentation::ActiveRecord` and
+`OpenTelemetry::Instrumentation::PG` instrumentation.
+With this setup, the EXPLAIN Plan span from the pganalyze collector will be
+created as a child trace of the ActiveRecord span, making it easy to relate the
+EXPLAIN Plan span to the ActiveRecord code.
+However, the span from the pg instrumentation will be created as a sibling span
+of the pganalyze collector, which may make it harder to relate them together.
+This is due to the limitation of when we can generate a query comment, which is
+currently only possible after the ActiveRecord instrumentation, but before the
+pg instrumentation.
```

Comment:
I still don't quite understand what this is telling me (this is probably due to my lack of familiarity with otel at this point). Specifically, what does it mean that it's created as a sibling span, and can I do anything about that? Also, is this a consequence of using the `OpenTelemetry::Instrumentation` classes, or a separate problem. If the latter should this be two separate limitations bullet points?

No need to edit this further on my account at this point (given I have no relevant otel experience), but I wanted to note some parts that are still unclear to me.

Reply:
FYI: I chatted with Maciek and explained this. We agreed that this is not needed as it's kinda small detail, so I'll drop this for now.